Add new icon as well as telemetry

chrdavis opened the pull request:
This change adds the new icon for PowerRename as well as some telemetry calls
- When we are invoked
- Was the invocation successful or not
- Enabled/Disabled set state of PowerRename
- Was the UI stood up successfully
- Details of the rename operation
  - total item count
  - selected item count
  - rename item count
  - flags used
  - list of extensions used with count

chrdavis committed:
Update icon and add telemetry calls

This change adds the new PowerRename icon and includes telemetry calls

chrdavis committed:
Ensure string is freed

enricogior commented on src/modules/powerrename/lib/trace.cpp:
```diff
@@ -0,0 +1,73 @@
+{
+  TraceLoggingWrite(
+        g_hProvider,
+        "PowerRename::Event::Invoked",
```
Hi Chris,
we are in the process of fixing the telemetry event names https://github.com/microsoft/PowerToys/pull/545
As you can see we are currently removing `::Event::` and using the underscore as a separator.

chrdavis committed:
Update event naming convention

chrdavis committed:
Delete PowerRenameDemo.gif

Deleting this gif as I added it with a separate commit with the README.md update for PowerRename

enricogior reviewed (approved):
LGTM!

CZEMacLeod commented:
The PowerRename.ico doesn't seem to show in the windows explorer context menu, only in the actual rename dialog window. It would be nice if it were in both places to be consistent and make it easy to find the option in the list. (some of us have a _lot_ of entries in our right click!)

chrdavis commented:
Good point.  I can make that change.

I'm also planning on adding a setting to move the context menu entry for PowerRename to the extended context menu only (SHIFT + Right-Click).  Most people use PowerRename infrequently so it shouldn't clutter the context menu all of the time.

CZEMacLeod commented:
@chrdavis It would be nice if PowerRename showed up _instead_ of Rename, if you had multiple files selected. 
The default Rename option seems to go to edit mode for the last file selected, which isn't a lot of use in my opinion. In this case (multiple files) it would be nice if it hooked the F2 hotkey too ;)

enricogior commented:
Hi @CZEMacLeod
I suspect we can't replace the default Rename, but we can for sure add a shortcut key.
Please open an issue to track the proposal.
Thank you.
